sway-core/src/semantic_analysis/namespace/trait_map.rs

```diff
@@ -853,12 +853,13 @@ impl TraitMap {
 
     /// Checks to see if the trait constraints are satisfied for a given type.
     pub(crate) fn check_if_trait_constraints_are_satisfied_for_type(
-        &self,
+        &mut self,
         handler: &Handler,
         type_id: TypeId,
         constraints: &[TraitConstraint],
         access_span: &Span,
         engines: &Engines,
+        try_inserting_trait_impl_on_failure: bool,
```

Comment:
This needs something more explicit than a boolean. It's dangerous to get wrong and not at all explicit from the callee side.